katello-attach-subscription

```diff
@@ -907,13 +896,37 @@ def vdcupdate()
       else
         virtualhostregex = nil
       end
+      # test all the facts
+      if @options[:verbose]
+        puts "VERBOSE: Start testing facts value for #{sys['name']}"
+      end
+      fact_test_passed = compareAllFacts(sub, sys)
```

Comment:
why don't you use `KatelloAttachSubscription::HostMatcher.match_host(sys, sub)` here?